### PR TITLE
Remove export of ScalarValues and VectorValues

### PR DESCRIPTION
**Pull request by KnutAM**
`ScalarValues` and `VectorValues` are exported but don't seem to be defined.
I also added a test to check that no symbols are exported that haven't been defined. Not sure where to put this test though...

### CODE REVIEW

**Comment by fredrikekre** [comment]
Removed in https://github.com/Ferrite-FEM/Ferrite.jl/pull/404. Docs look more ugly now, it was nice to be able to reference both `CellVectorValues` and `FaceVectorValues` as `VectorValues`:P

**Comment by KnutAM** [comment]
Not sure why "curl" fails, the new test passes when I run `]test` on my computer

**Comment by kimauth** [comment]
We could document the `VectorValued` and `ScalarValued` Traits and refer to them in the docs instead. Though the question is where in the docs that should go, it's not really supposed to be exposed to the user.

Not sure if I like writing out `CellScalarValues` and `FaceScalarValues` everywhere, e.g. `PointScalarValues` uses some of these functions as well. That list easily becomes lengthy.

File: test/runtests.jl [hunk 1]
```diff
@@ -22,3 +22,4 @@ include("test_pointevaluation.jl")
 # include("test_notebooks.jl")
 include("test_apply_rhs.jl")
 include("test_examples.jl")
+@test isa([Core.eval(Ferrite,type) for type in names(Ferrite)], Vector)  # Test that all exported symbols are defined
```
Comment (fredrikekre) [inline test/runtests.jl:25]: I think there is a standard way of testing this, but can't find it right away.

**Comment by KnutAM** [comment]
Could one write e.g "where `u` is [`VectorValued`](https://github.com/Ferrite-FEM/Ferrite.jl/pull/404)" where the link is to a page documenting what VectorValued means (but not giving this page in the menu). And also documenting `ScalarValued` on that page?

File: test/runtests.jl (hunk 1 above)
Comment (KnutAM) [reply test/runtests.jl:25]: https://github.com/JuliaLang/julia/issues/18780?

Comment (fredrikekre) [reply test/runtests.jl:25]: No, I was thinking of https://github.com/JuliaTesting/Aqua.jl/blob/master/src/exports.jl perhaps, but that seems a bit overkill, perhaps
```suggestion
@test all(x -> isdefined(Ferrite, x), names(Ferrite))  # Test that all exported symbols are defined

```
is sufficient.

**Comment by kimauth** [comment]
Seems not so nice to me to hide it away like that. Perhaps just an introductory section at the top of https://ferrite-fem.github.io/Ferrite.jl/stable/reference/fevalues/ which explains that there are `ScalarValues` and `VectorValues`, where the `VectorValues` are zero-padded versions of the `ScalarValues`? I remember that I found this hard to understand in the beginning.

**Comment by KnutAM** [comment]
Yes, that would be even better!

**Comment by kimauth** [comment]
No need to do it in this PR, the PR is after all about removing the export that I forgot to remove in #404 . I can draft something for the `VectorValues` and `ScalarValues`.

**Review by kimauth — APPROVED** [review]

**Pull request merged**